cf_xarray/accessor.py

```diff
@@ -403,31 +429,33 @@ def get_single(obj: Union[DataArray, Dataset], key: str):
             raise KeyError(f"No results found for {key!r}.")
         return results
 
+    get_single.__doc__ = func.__doc__.replace("One or more of", "One of")
+
     return get_single
 
 
 #: Default mappers for common keys.
 _DEFAULT_KEY_MAPPERS: Mapping[str, Tuple[Mapper, ...]] = {
-    "dim": (_get_dims,),
-    "dims": (_get_dims,),  # transpose
-    "drop_dims": (_get_dims,),  # drop_dims
-    "dimensions": (_get_dims,),  # stack
-    "dims_dict": (_get_dims,),  # swap_dims, rename_dims
-    "shifts": (_get_dims,),  # shift, roll
-    "pad_width": (_get_dims,),  # shift, roll
+    "dim": (_dims(_get_all),),
```

Comment:
How about using more decorators? Too messy?

Reply:
i like your odler version with `_get_dims` etc. that's nice and clean. And `_single` can be the only decorator since it is not used very frequently. `_variables` could also be a decorator I guess.

Reply:
Yup, I was very undecided about it!